app/premium/_components/checkoutButton.tsx

```diff
@@ -1,30 +1,38 @@
 'use client'
 
-import { Button } from '@/components/ui/button'
+import { Button, type ButtonProps } from '@/components/ui/button'
+import { Fragment, useEffect, useState } from 'react'
 import { loadStripe } from '@stripe/stripe-js'
-import { useState } from 'react'
+import { LoadingSvg } from '@/icons/loading'
```

Comment:
Lets replace this with the lucide icon, looks better imo and matches the rest of the apps icons.
```suggestion
import { LoaderCircle } from 'lucide-react'
```